mutable_tree.go

```diff
@@ -56,7 +56,7 @@ func NewMutableTree(db dbm.DB, cacheSize int, skipFastStorageUpgrade bool, lg lo
 	}
 
 	ndb := newNodeDB(db, cacheSize, opts, lg)
-	head := &ImmutableTree{ndb: ndb, skipFastStorageUpgrade: skipFastStorageUpgrade}
+	head := &ImmutableTree{ndb: ndb, skipFastStorageUpgrade: skipFastStorageUpgrade, version: int64(opts.InitialVersion)}
```

Comment:
I know the result is same.
The happy case is to load the version while constructing the tree in `LoadVersion`, so we don't need to set the `InitialVersion` option. The `InitialVersion` is just optional considering the above case like new module.
But if we force set the version here, it can be misleading the meaning of `InitialVersion`.

Reply:
yea sure, I don't have strong opinion about this.
I will update PR after testing :)

Reply:
This fix is also working well 👍

Reply:
I don't have strong opinion on this, so happy to follow your way. 

Is the above solution (in comment) is good? or should we make other changes on `LoadVersion` function?

Reply:
no, we should move this part to the `LoadVersion`, your commit will be not working when set the initial version by `SetInitialVersion` without opts.

Reply:
okay, I moved the fix to LoadVersion and tested it. It's also working well :)